rewrite-core/src/main/java/org/openrewrite/text/CreateTextFile.java

```diff
@@ -92,13 +95,17 @@ public Collection<SourceFile> generate(AtomicBoolean shouldCreate, ExecutionCont
     @Override
     public TreeVisitor<?, ExecutionContext> getVisitor(AtomicBoolean created) {
         Path path = Paths.get(relativeFileName);
-        return new PlainTextVisitor<ExecutionContext>() {
+        return new TreeVisitor<SourceFile, ExecutionContext>() {
             @Override
-            public PlainText visitText(PlainText text, ExecutionContext ctx) {
-                if ((created.get() || Boolean.TRUE.equals(overwriteExisting)) && path.toString().equals(text.getSourcePath().toString())) {
-                    return text.withText(fileContents);
+            public @Nullable SourceFile visit(@Nullable Tree tree, ExecutionContext executionContext) {
+                SourceFile sourceFile = (SourceFile) requireNonNull(tree);
+                if (sourceFile instanceof Quark || sourceFile instanceof Remote || sourceFile instanceof Binary) {
+                    return sourceFile;
+                }
+                if ((created.get() || Boolean.TRUE.equals(overwriteExisting)) && path.toString().equals(sourceFile.getSourcePath().toString())) {
+                    return PlainTextParser.convert(sourceFile).withText(fileContents);
```

Comment:
Since you're wholly overwriting the existing file then there's no need to use `PlainTextParser.convert()` since you aren't interacting with its contents in any way. You can just new up a `PlainText` with an appropriate source path. Then you don't even need the guard against quark/binary/remote.

Reply:
It would be nice if we had a registry of our parsers so that we could parse the source. But since we don't have access to the compile-time environment at this point, it would at best end up producing a not properly type-attributed LST.